### PR TITLE
Update CHANGELOG.md with v0.10.1 release

### PR DESCRIPTION
**Pull request by amartinezfayo**
Update CHANGELOG.md with v0.10.1 release.
The release candidate commit is 6ea1530.
Release date is tentatively set to 2020-06-19. If needed, the PR will be updated with the final date.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## [0.10.1] - 2020-06-19
+- Added Upstream Authority `vault` Plugin (#1611, #1632)
+- Improved cgroup path parsing in the `docker` workload attestor plugin (#1605)
+- Improved container ID parsing in the `k8s` workload attestor plugin (#1649)
```
Comment (azdagron) [inline CHANGELOG.md:6]: These should be worded similarly since they both involve parsing the container ID from the cgroup path

Comment (azdagron) [reply CHANGELOG.md:6]: (I recognize the PRs themselves didn't have consistent wording :) )

**Comment by anvega** [comment]
Thanks @amartinezfayo. As discussed over slack it would be great to breakdown the list into two categories; what's new, and what's changed. 

Something along the lines of: 

### What's New
- Add Upstream Authority `Vault` Plugin (#1611, #1632)
- Improved configuration file docs to list all possible configuration settings (#1608, #1618)
### What's changed
- Improved container ID parsing from cgroup path in the `docker` workload attestor plugin (#1605)
- Improved container ID parsing from cgroup path in the `k8s` workload attestor plugin (#1649)
- Envoy SDS support is now always on (#1579)
- Errors on agent SVID rotation are now fatal if the agent's current SVID has expired, forcing an agent restart (#1584)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## [0.10.1] - 2020-06-19
+- Added Upstream Authority `vault` Plugin (#1611, #1632)
```
Comment (anvega) [inline CHANGELOG.md:4]:
```suggestion
- 'vault' as Upstream Authority built-in plugin(#1611, #1632)
```

**Pull request merged**